hgt/index.js

```diff
@@ -1,39 +1,31 @@
-const fs = require('fs');
+const { S3Client, GetObjectCommand } = require('@aws-sdk/client-s3')
 const THREE_ARC_SECOND = 1442401 * 2;
 const ONE_ARC_SECOND = 12967201 * 2;
 
+const s3Client = new S3Client({region: process.env.AWS_REGION || 'us-east-1'});
+
 // Adapted from https://github.com/perliedman/node-hgt/blob/master/src/hgt.js
 function HGT(path, swLngLat, options, callback) {
-    // Make sure the file exists
-    fs.open(path, 'r', (error, fd) => {
-        setImmediate(() => {
-            if (error) return callback(error);
-
-            // Determine the type of HGT file based on the size of the file
-            fs.fstat(fd, (error, stats) => {
-                setImmediate(() => {
-                    if (error) return callback(error);
-                    const [resError, resAndSize] = getResolutionAndSize(
-                        stats.size
-                    );
-                    if (resError) return callback(resError);
+    s3Client.send(new GetObjectCommand({
+        Bucket: process.env.AWS_ELEVATION_BUCKET,
+        Key: path
+    })).then(async (dem) => {
+        const [resError, resAndSize] = getResolutionAndSize(
+            dem.ContentLength
+        );
+        if (resError) return callback(resError);
 
-                    // Stream the file contents to a Buffer
-                    getHGTBuffer(path, (error, buffer) => {
-                        setImmediate(() => {
-                            if (error) return callback(error);
-                            callback(undefined, {
-                                buffer,
-                                resolution: resAndSize.resolution,
-                                size: resAndSize.size,
-                                options,
-                                swLngLat,
-                            });
-                        });
-                    });
-                });
-            });
+        const buffer = await streamToBuffer(dem.Body)
+        return callback(undefined, {
+            buffer,
+            resolution: resAndSize.resolution,
+            size: resAndSize.size,
+            options,
+            swLngLat,
         });
+    }).catch((error) => {
+        console.log(`Error fetching tile ${path}`, error)
+        callback(`Error fetching tile ${path}`)
```

Comment:
double-checking: can the callback handle passing the error arg as just a string? I see we pass the caught error in other callback() invocations.

Reply:
Indeed. The type of the callback arg doesn't matter so much as the presence of it